core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -790,8 +790,12 @@ protected LogContext getLogContext() {
         }
 
         public void run() {
-            if (isCancelled())
+            boolean isReady = remoteLogMetadataManager.isReady(topicIdPartition);
+            if (isCancelled() || !isReady) {
+                logger.debug("Skipping the current run for partition {} as it is either cancelled: {} or " +
+                        "remote log metadata is not ready, isReady: {}", topicIdPartition, isCancelled(), isReady);
                 return;
+            }
```

Comment:
Wouldn't be better to have two separate if statements here?

Reply:
Updated.